RFCs/0011-partiql-insert-update-spec.md

```diff
@@ -0,0 +1,845 @@
+
+For the following examples we assume the existence of composite key `partition_key_sort_key` comprising `HK` and `RK` attributes i.e `HK_RK`.
+
+##### Example 4.2.7
+
+The following example leads to a `SemanticError` exception because `sort_key` is absent from the value specified by the `REPLACE`.
+
+```SQL
+-- Existing Item is: 
+--  {HK: 1, RK: 1, myAttr: 12}
+-- Outcome is a `SemanticError` with the existing item being intact.
+
+INSERT into Customers
+<<{HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello"}>>
+ON CONFLICT
+    DO REPLACE {HK: 1, thirdAttr: "World"};
+```
+
+##### Example 4.2.8
+
+The following example leads to a `SemanticError` exception because partition key is absent from the value specified by replace.
+
+```SQL
+-- Existing Item is {HK: 1, RK: 1, myAttr: 12}
+-- Outcome is a `SemanticError` with the existing item being intact.
+
+INSERT into Customers
+<<{HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello"}>>
+ON CONFLICT
+    DO REPLACE {RK: 1, thirdAttr: "World"};
+```
+
+##### Example 4.2.9
+
+The following example leads to a `SemanticError` exception because both partition key and sort_key are absent in the update statement:
+
+```SQL
+-- Existing Item is { HK: 1, RK: 1, myAttr: 12 }
+-- Outcome is a SemanticError
+INSERT into Customers
+<<{HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello"}>>
+ON CONFLICT
+    DO REPLACE {thirdAttr: "World"};
+```
+
+##### Example 4.2.10
+
+The following example leads to replacement of the item specified by replace with the existing one:
+
+```SQL
+-- Existing Item is:
+--  {HK: 1, RK: 1, myAttr: 12 }
+-- Outcome is:
+--  {HK: 1, RK: 1, thirdAttr: "World"}
+
+INSERT into Customers
+<<{HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello"}>>
+ON CONFLICT
+    DO REPLACE {HK: 1, RK: 1, thirdAttr: "World"};
+```
+
+##### Example 4.2.11
+
+The following example SHOULD* lead to replacing the existing item with the item specified by `REPLACE` as a new item:
+
+```SQL
+-- Existing items are:
+--  {HK: 1, RK: 1, myAttr: 12 }, 
+--  {HK: 1, RK: 2, myAttr: 12 }
+-- Outcome:
+--  {HK: 1, RK: 3, thirdAttr: “World”}, 
+--  {HK: 1, RK: 2, myAttr: 12 }
+
+INSERT into Customers
+<<{HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello" }>>
+ON CONFLICT
+    DO REPLACE {HK: 1, RK: 3, thirdAttr: "World"};
+```
+
+** SHOULD, because for a distributed database system, implementing this logic may lead to data inconsistency. This is because there is no guarantee that the INSERT procedure that generates the conflict executes on the same database host where the conflict is going to get resolved.
+
+##### Example 4.2.12
+
+The following example MUST lead to a `SemanticError` if there is an existing item that conflicts with the item specified by replace:
+
+```SQL
+-- Existing items is:
+--  {HK: 1, RK: 1, myAttr: 12 },
+--  {HK: 1, RK: 2, myAttr: 12 }
+-- Outcome is SemanticError
+
+INSERT into Customers
+<<{ HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello" }>>
+ON CONFLICT
+    DO REPLACE {HK: 1, RK: 2, thirdAttr: "World"};
```

Comment:
Is there a valid example of `DO REPLACE EXCLUDED`?

Reply:
Also doesn't the grammar dictate that it is `DO REPLACE SET {HK: 1, RK: 2, thirdAttr: "World"}`?